src/main/java/org/apache/cassandra/sidecar/cdc/CDCBookmark.java

```diff
@@ -0,0 +1,213 @@
+package org.apache.cassandra.sidecar.cdc;
+
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.locks.ReentrantLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.db.commitlog.CommitLogDescriptor;
+import org.apache.cassandra.db.commitlog.CommitLogPosition;
+import org.apache.cassandra.io.util.DataInputPlus;
+import org.apache.cassandra.io.util.DataOutputBuffer;
+import org.apache.cassandra.io.util.DataOutputPlus;
+import org.apache.cassandra.sidecar.Configuration;
+
+/**
+ * Manages the CDC reader bookmark. This tracks the last successfully processed offset
+ * of a commit log.
+ */
+@Singleton
+public class CDCBookmark extends TimerTask
+{
+    // Tracks last disk sync'd commit log position.
+    private CommitLogPosition lastSyncedPosition;
+    // Tracks last successfully processed commit log position by the CDC reader.
+    private CommitLogPosition lastProcessedPosition;
+    private final Timer timer;
+    private static final String BOOKMARK = "CdcReader.bookmark";
+    private static final Logger logger = LoggerFactory.getLogger(CDCBookmark.class);
+    private final ReentrantLock bookmarkLock = new ReentrantLock();
+    private final Configuration conf;
+
+    @Inject
+    CDCBookmark(Configuration conf)
```

Comment:
Configuration conf is not used in  the class, so I want to know if we will use in the future ?same with MutationHandler class

Reply:
+1